packages/hierarchy-builder/src/test/internal/operators/grouping/PropertiesGrouping.test.ts

```diff
@@ -0,0 +1,1482 @@
+              label: "PropertyValue1",
+              key: expectedGroupingNodeKey1,
+              children: [{ ...nodes[0], parentKeys: [...nodes[0].parentKeys, expectedGroupingNodeKey1] }],
+            }),
+            createTestProcessedGroupingNode({
+              label: "PropertyValue2",
+              key: expectedGroupingNodeKey2,
+              children: [{ ...nodes[1], parentKeys: [...nodes[1].parentKeys, expectedGroupingNodeKey2] }],
+            }),
+          ],
+          ungrouped: [],
+        });
+      });
+
+      it("groups only nodes which match the provided property name", async () => {
+        const nodes = [
+          createTestProcessedInstanceNode({
+            key: { type: "instances", instanceKeys: [{ className: "TestSchema.Class", id: "0x1" }] },
+            processingParams: {
+              grouping: {
+                byProperties: {
+                  fullClassName: "TestSchema.Class",
+                  propertyGroups: [{ propertyName: "PropertyName1", propertyValue: "PropertyValue" }],
+                },
+              },
+            },
+          }),
+          createTestProcessedInstanceNode({
+            key: { type: "instances", instanceKeys: [{ className: "TestSchema.Class", id: "0x2" }] },
+            processingParams: {
+              grouping: {
+                byProperties: {
+                  fullClassName: "TestSchema.Class",
+                  propertyGroups: [{ propertyName: "PropertyName2", propertyValue: "PropertyValue" }],
+                },
+              },
+            },
+          }),
+        ];
+        const ecClass = { fullName: "TestSchema.Class" } as unknown as ECClass;
+        const property = { name: "PropertyName1", isPrimitive: () => true } as unknown as ECProperty;
+        classStubs.stubEntityClass({
+          schemaName: "TestSchema",
+          className: "Class",
+          is: async () => true,
+          properties: [property],
+        });
+        const propertyInfo: propertiesGrouping.PropertyGroupInfo = {
+          ecClass,
+          previousPropertiesGroupingInfo: [],
+          propertyGroup: { propertyName: "PropertyName1" },
+        };
+        const expectedGroupingNodeKey: GroupingNodeKey = {
+          type: "property-grouping:value",
+          propertyName: "PropertyName1",
+          fullClassName: "TestSchema.Class",
+          formattedPropertyValue: "PropertyValue",
+        };
+        expect(await propertiesGrouping.createPropertyGroups(metadataProvider, nodes, propertyInfo, formatter)).to.deep.eq({
+          groupingType: "property",
+          grouped: [
+            createTestProcessedGroupingNode({
+              label: "PropertyValue",
+              key: expectedGroupingNodeKey,
+              children: [{ ...nodes[0], parentKeys: [...nodes[0].parentKeys, expectedGroupingNodeKey] }],
+            }),
+          ],
+          ungrouped: [nodes[1]],
+        });
+      });
+    });
+
+    describe("Range is provided", async () => {
```

Comment:
- I think this shouldn't be nested under `propertyInfo is different from nodes' byProperties processing params` describe block
- I think it should be renamed to `range grouping`